app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -124,10 +124,16 @@ class PersonalDetailsCollectionController @Inject()(personalDetailsSubmission: P
 
   def whatIsYourNino(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
     viewConfig.isLoggedIn.flatMap { isLoggedIn: Boolean =>
-      Future.successful(Ok(what_is_your_nino(ninoForm, completionUrl, isLoggedIn, failureUrl)))
+      sessionKeyCheck(
+        Future.successful(Ok(what_is_your_nino(ninoForm, completionUrl, isLoggedIn, failureUrl))),
+        completionUrl,
+        failureUrl
+      )
     }
   }
 
+
+
   def submitYourNino(completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl]): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
can we use your function in this method aswell? it will make it cleaner and will guard against multiple tabs being open (L145 -> 152).
we don't need to return BadRequest, we can just use your function OOTB